web/src/js/components/Modes/Popover.tsx

```diff
@@ -0,0 +1,37 @@
+import * as React from "react";
+
+interface PopoverProps {
+    children: React.ReactNode;
+}
+
+export function Popover({ children }: PopoverProps) {
+    const id = React.useId();
+
+    // Rather annoying workaround to make popover anchors work with current React.
+    // Ideally this can go away once browsers have anchor-scope.
+    // https://github.com/facebook/react/issues/26839
+    const cssId =
+        "--" + [...id].map((c) => c.charCodeAt(0).toString(16)).join("");
+    const buttonRef = React.useRef<HTMLButtonElement>(null);
+    const popoverRef = React.useRef<HTMLDivElement>(null);
+    React.useEffect(() => {
+        // @ts-expect-error no anchor support yet
+        buttonRef.current!.style.anchorName = cssId;
+        // @ts-expect-error no anchor support yet
+        popoverRef.current!.style.positionAnchor = cssId;
+    }, []);
```

Comment:
Really nice! :) I tested it, and it works as expected! It took me a while to understand what was going on here 😅

Reply:
That's an indication that we need an explanatory comment! :)